addons/account/models/ir_actions_report.py

```diff
@@ -23,7 +23,12 @@ def _post_pdf(self, save_in_attachment, pdf_content=None, res_ids=None):
             res_ids = None
             if not save_in_attachment:
                 raise UserError(_("No original vendor bills could be found for any of the selected vendor bills."))
-        return super(IrActionsReport, self)._post_pdf(save_in_attachment, pdf_content=pdf_content, res_ids=res_ids)
+        res = super()._post_pdf(save_in_attachment, pdf_content, res_ids)
+        if self.model == 'account.move':
+            moves = self.env[self.model].browse(res_ids)
+            for move in moves.filtered(lambda m: m.type == 'out_invoice'):
+                self.retrieve_attachment(move).register_as_main_attachment(force=False)
```

Comment:
I think it doesn't work since the attachment is created in `postprocess_pdf_report`.
Also, you need to use `moves.filtered(lambda m: m.is_sale_document(include_receipts=True)` instead